File: holiday_warmups/formatWithPadding.js
const formatWithPadding = (numInput, char, numLength) => {
  numInputStr = numInput.toString();
  numInputArray = numInputStr.split("");
  //   console.log(numInputStr);
  //   console.log(numInputArray);
  //   console.log(numInputArray.length);

  //   console.log(numInputStr.length);
  outputArray = [];
  if (numInputStr.length >= Number(numLength)) {
    for (let i = 0; i < numInputStr.length; i++) {
      outputArray.push(numInputStr[i]);
      //   console.log(outputArray);
    }
    return outputArray.join("");
  } else {
    for (let j = numInputArray.length; j <= Number(numLength - 1); j++) {
      numInputArray.unshift(char);
    }
    return numInputArray.join("");
  }
};

console.log(formatWithPadding(1234, "*", 3));
console.log(formatWithPadding(123, "0", 5)); //=> "00123"
console.log(formatWithPadding(42, "*", 10)); //=> "********42"

File: holiday_warmups/formatWithPadding.rb


def format_with_padding (num_input,char,num_length) 

    num_input_array = num_input.to_s
    # p num_input_array
    # p num_input_array.length
    num_output_array = []
    num_length_array = num_length.to_i
    # p num_length_array
    if (num_input_array.length >= num_length_array)
        
        return num_input.to_s
    else
        counter = num_input_array.length
        # p counter
        num_output_array = num_input_array.split('')
        # p num_output_array
        while counter < num_length_array
            num_output_array.unshift(char)
            counter += 1
        end
        return num_output_array.join('')
    end
    

end
p format_with_padding(1234, '*', 3); # "1234"
p format_with_padding(123, '0', 5); # "00123"
p format_with_padding(42, '*', 10); # "********42"